bakery/checkers/time.go

```diff
@@ -30,3 +32,38 @@ var TimeBefore = CheckerFunc{
 func TimeBeforeCaveat(t time.Time) Caveat {
 	return firstParty(CondTimeBefore, t.UTC().Format(time.RFC3339Nano))
 }
+
+// ExpireTime finds any caveats created by TimeBeforeCaveat and
+// calculates the time at which the macaroon will no longer be valid. If
+// there are no such caveats expires will be false, otherwise expires is true.
+func ExpireTime(cavs []macaroon.Caveat) (t time.Time, expires bool) {
+	for _, cav := range cavs {
+		if strings.HasPrefix(cav.Id, CondTimeBefore) {
+			et, err := time.Parse(CondTimeBefore+" "+time.RFC3339Nano, cav.Id)
+			if err != nil {
+				continue
+			}
+			if t.IsZero() || t.After(et) {
```

Comment:
I think I'd use et.Before(t) because it's more usual to have the
value just created on the LHS.